### PR TITLE
"You'll find all my treasure in one place..." -- Adds optional last words whispered when succumbing

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Adds a dialog for trying to say your last words when you succumb from crit. 
Also allows you to succumb when you're below -100 HP, which might not have actually been possible before?

If you're in losebreath, this won't do much for you, so I'm hoping that https://github.com/ParadiseSS13/Paradise/pull/23425 goes in before this one.

## Why It's Good For The Game
It adds a bit of fun flavor to death, and lets you choose how you'd like to go out. You can summon just enough energy to break out of unconsciousness for one last phrase...and then you're done.

## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/89928798/18c8631d-88ae-477a-9922-f41b231d4be8

## Testing

Punched myself to death, then tried the succumb verb.

## Changelog
:cl:
add: You can now utter some last words when you succumb to death from crit.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -288,20 +288,34 @@
...
-		// super check for weird mobs, including ones that adjust hp
-		// we don't want to go overboard and gib them, though
-		for(var/i = 1 to 5)
-			if(health < HEALTH_THRESHOLD_DEAD)
-				break
-			take_overall_damage(max(5, health - HEALTH_THRESHOLD_DEAD), 0)
-		death()
-		to_chat(src, "<span class='notice'>You have given up life and succumbed to death.</span>")
+	set hidden = TRUE
+	if(!can_succumb())
```
Comment (Contrabang) [inline code/modules/mob/living/living.dm:297]: I think this should be simplified, I can't think of anywhere else can_succumb should be used

Comment (lewcc) [reply code/modules/mob/living/living.dm:297]: ough I kinda meant to get rid of that

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -288,13 +288,9 @@
...
 	return TRUE
 
-/mob/living/proc/can_succumb()
-	return health < HEALTH_THRESHOLD_CRIT && stat == UNCONSCIOUS
-
-
 /mob/living/verb/succumb()
 	set hidden = TRUE
-	if(!can_succumb())
+	if(health >= HEALTH_THRESHOLD_CRIT || stat != UNCONSCIOUS)
```
Comment (Contrabang) [inline code/modules/mob/living/living.dm:293]: you should have a (stat == dead) return before this properly so that the message only shows if you're alive

**Comment by ItsMarmite** [comment]
Does this work over radio?

**Comment by Contrabang** [comment]
> Does this work over radio?

nope

**Comment by S34NW** [comment]
Please merge `upstream/master` into your branch to fix CI failing due to external factors.

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -289,19 +289,36 @@
...
+		create_log(MISC_LOG, "gave their final words, [last_words]")
+		whisper(last_words)
+
+	add_attack_logs(src, src, "[src] has [!isnull(last_words) ? "whispered [p_their()] final words" : "succumbed to death"] with [round(health, 0.1)] points of health!")
+
+	create_log(MISC_LOG, "has succumbed to death with [round(health, 0.1)] points of health")
+	adjustOxyLoss(max(health - HEALTH_THRESHOLD_DEAD, 0))
+	// super check for weird mobs, including ones that adjust hp
+	// we don't want to go overboard and gib them, though
+	for(var/i = 1 to 5)
```
Comment (Sirryan2002) [inline code/modules/mob/living/living.dm:313]:
```suggestion
	for(var/i in 1 to 5)
```

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]
Needs balance team approval in addition to existing design team approval

**Comment by spacegeese** [comment]
how do you do it without typing succumb manually into the bar?

**Pull request merged**